assets/php/class-wgpb-block-library.php

```diff
@@ -0,0 +1,254 @@
+		self::register_script( 'wc-product-top-rated', plugins_url( 'build/product-top-rated.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-products-attribute', plugins_url( 'build/products-attribute.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-featured-product', plugins_url( 'build/featured-product.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+	}
+
+	/**
+	 * Register blocks, hooking up assets and render functions as needed.
+	 */
+	public static function register_blocks() {
+		require_once dirname( __FILE__ ) . '/class-wgpb-block-featured-product.php';
+
+		register_block_type(
+			'woocommerce/handpicked-products',
+			array(
+				'editor_script' => 'wc-handpicked-products',
+				'editor_style'  => 'wc-block-editor',
+				'style'         => 'wc-block-style',
+			)
+		);
+		register_block_type(
+			'woocommerce/product-best-sellers',
+			array(
+				'editor_script' => 'wc-product-best-sellers',
+				'editor_style'  => 'wc-block-editor',
+				'style'         => 'wc-block-style',
+			)
+		);
+		register_block_type(
+			'woocommerce/product-category',
+			array(
+				'editor_script' => 'wc-product-category',
+				'editor_style'  => 'wc-block-editor',
+				'style'         => 'wc-block-style',
+			)
+		);
+		register_block_type(
+			'woocommerce/product-new',
+			array(
+				'editor_script' => 'wc-product-new',
+				'editor_style'  => 'wc-block-editor',
+				'style'         => 'wc-block-style',
+			)
+		);
+		register_block_type(
+			'woocommerce/product-on-sale',
+			array(
+				'editor_script' => 'wc-product-on-sale',
+				'editor_style'  => 'wc-block-editor',
+				'style'         => 'wc-block-style',
+			)
+		);
+		register_block_type(
+			'woocommerce/product-top-rated',
+			array(
+				'editor_script' => 'wc-product-top-rated',
+				'editor_style'  => 'wc-block-editor',
+				'style'         => 'wc-block-style',
+			)
+		);
+		register_block_type(
+			'woocommerce/products-by-attribute',
+			array(
+				'editor_script' => 'wc-products-attribute',
+				'editor_style'  => 'wc-block-editor',
+				'style'         => 'wc-block-style',
+			)
+		);
+		register_block_type(
+			'woocommerce/featured-product',
+			array(
+				'render_callback' => array( 'WGPB_Block_Featured_Product', 'render' ),
+				'editor_script'   => 'wc-featured-product',
+				'editor_style'    => 'wc-block-editor',
+				'style'           => 'wc-block-style',
+			)
+		);
+	}
+
+	/**
+	 * Adds a WooCommerce category to the block inserter.
+	 *
+	 * @param array $categories Array of categories.
+	 * @return array Array of block categories.
+	 */
+	public static function add_block_category( $categories ) {
+		return array_merge(
+			$categories,
+			array(
+				array(
+					'slug'  => 'woocommerce',
+					'title' => __( 'WooCommerce', 'woo-gutenberg-products-block' ),
+					'icon'  => 'woocommerce',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Output the wcSettings global before printing any script tags.
```

Comment:
Since this is formally introducing `wcSettings` into core, I wonder if we should add a filter here? So that we can target the filter instead and only add new/additional fields in `wc-admin`.

Reply:
I'm not sure I understand– so this would just be `$settings = apply_filters( 'woocommerce_components_settings', array() )` ? (maybe a different name, I haven't looked up WC hook conventions)

Reply:
Yeah that's what I meant -- sorry for not being clearer.

I was mainly thinking that we could rely on that filter in `wc-admin` to build upon what will be in core/3.6, instead of duplicating code for common things like data endpoints, stockStatuses, etc.

Reply:
I've just pushed another commit that adds the filter – I was hoping I could remove creating $settings, and just let the filter take care of it, but if wc-admin doesn't exist, we don't have the objects needed in the components – though if some of these can be removed from this initialization that would be good. Let me know if this looks ok now.